cdap-security/src/main/java/co/cask/cdap/security/store/FileSecureStore.java

```diff
@@ -264,6 +288,10 @@ private static KeyStore locateKeystore(Path path, final char[] password) throws
       if (Files.exists(path)) {
         loadFromPath(ks, path, password);
       } else {
+        Path parent = path.getParent();
```

Comment:
This creates the "secure" directory if it does not exist. The keystore creation fails if the parent is not present.